### PR TITLE
Media.js should be loaded in if media atoms are on the page

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?
Trello card: https://trello.com/c/Yc5YBuSa/397-web-load-mediajs-if-media-atoms-or-carousel-are-in-the-page

Media JS is loaded if there are video elements on the page, but it also has to be loaded if there are youtube atoms on the page. 

## What is the value of this and can you measure success?
Video carousels will work on sport fronts now (sport uses entirely youtube now)

## Does this affect other platforms - Amp, Apps, etc?
Nope

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

Nope

## Screenshots
N/A

## Tested in CODE?
Tested on a code front locally

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/main.js [hunk 1]
```diff
@@ -179,7 +179,9 @@ const bootEnhanced = (): void => {
     }
 
     if (
-        (config.isMedia || qwery('video, audio').length) &&
+        (config.isMedia ||
```
Comment (NataliaLKB) [inline static/src/javascripts/bootstraps/enhanced/main.js]: Prettier decided this was prettier :woman_shrugging:

File: static/src/javascripts/bootstraps/enhanced/main.js [hunk 2]
```diff
@@ -179,7 +179,9 @@ const bootEnhanced = (): void => {
...
+            qwery('video, audio').length ||
+            $('.youtube-media-atom').length > 0) &&
```
Comment (gustavpursche) [inline static/src/javascripts/bootstraps/enhanced/main.js]: Oh wow. This is an expensive switch! Why not combining both of them into one query `qwery('video, audio, .youtube-media-atom').length`? We should also wrap this into fastdom, if it's available at this point.

What do you think?

Comment (NataliaLKB) [inline static/src/javascripts/bootstraps/enhanced/main.js]: done!

**Comment by gustavpursche** [comment]
@NataliaLKB Close to what I meant ;)

```js
if (!config.page.isHosted) {
    fastdom.read(() => qwery('video, audio, .youtube-media-atom'))
      .then(els => {
        if (config.isMedia || els.length) {
          require.ensure(
              [],
              require => {
                  bootstrapContext(
                      'media',
                      require('bootstraps/enhanced/media/main').init
                  );
              },
              'media'
          );
        }
      });
}
```

Why?

- `!config.page.isHosted` is very cheap
- `fastdom.read(() => qwery('video, audio, .youtube-media-atom'))` because we can't make sure, `bootstraps/enhanced/media/main` doesn't perform a write operation and therefore it shouldn't be wrapped by `.read()`

**Comment by NataliaLKB** [comment]
Makes sense @gustavpursche ! All done.

**Review by gu-stav — APPROVED** [review]
Great, thanks! @SiAdcock might have a second look, just to have another brain thinking about any issue that could come up.

**Review by SiAdcock — APPROVED** [review]
Clever solution, nice one @NataliaLKB and @gustavpursche!

**Review by akash1810 — APPROVED** [review]
awesome!

img

**Pull request merged**